lib/whitehall_importer/create_edition.rb

```diff
@@ -21,15 +21,61 @@ def call
 
       edition = if whitehall_edition["state"] == "withdrawn"
                   create_withdrawn_edition
+                elsif edited_unpublishing?
+                  split_unpublished_edition
+                elsif unpublished_edition?
+                  create_removed_edition
                 else
-                  create_edition
+                  state = MigrateState.call(whitehall_edition["state"], whitehall_edition["force_published"])
+                  status = build_status(state)
+                  create_edition(status: status, current: current, edition_number: edition_number)
                 end
 
       edition.tap { |e| access_limit(e) }
     end
 
   private
 
+    def revision
+      @revision ||= CreateRevision.call(document, whitehall_edition)
+    end
+
+    def history
+      @history ||= EditionHistory.new(whitehall_edition["revision_history"])
+    end
+
+    def unpublished_edition?
+      whitehall_edition["unpublishing"] && %w[submitted rejected draft].include?(whitehall_edition["state"])
+    end
+
+    def edited_unpublishing?
+      return false unless whitehall_edition["revision_history"].second_to_last
+
+      whitehall_edition["revision_history"].second_to_last["state"] != "published" && unpublished_edition?
```

Comment:
Should this be a question we ask the history object?

Reply:
Moved to EditionHistory 👍